runtime/near-vm-runner/src/preload.rs

```diff
@@ -91,29 +140,48 @@ impl ContractCaller {
                 match call_data.result {
                     Err(err) => (None, Some(err)),
                     Ok(module) => match module {
-                        Wasmer0(module) => run_wasmer0_module(
-                            module,
-                            method_name,
-                            ext,
-                            context,
-                            vm_config,
-                            fees_config,
-                            promise_results,
-                            profile,
-                            current_protocol_version,
-                        ),
-                        Wasmer1(module, store) => run_wasmer1_module(
-                            &module,
-                            &store,
-                            method_name,
-                            ext,
-                            context,
-                            vm_config,
-                            fees_config,
-                            promise_results,
-                            profile,
-                            current_protocol_version,
-                        ),
+                        Wasmer0(module) => match &mut self.vm_data_private {
+                            Some(data) => match data {
```

Comment:
I think you can match on tupples here:

```
match (module, &mut self.vm_data_private) {
  (Wasmer0(module), VMDataPrivate::Wasmer0(memory)) => { ... }
  (Wasmer1(module), VMDataPrivate::Wasmer1(memory)) => { ... }
  _ => panic!(),
}
```

I don't think there's a super-good way to remove the unreachable state altogether here.